src/main/java/lv/ctco/javaschool/goal/control/GoalStore.java

```diff
@@ -19,12 +22,26 @@
     private EntityManager em;
 
     public List<Goal> getGoalsListFor(User user) {
-        return em.createQuery(
+        List<Goal> goals = em.createQuery(
                 "select g " +
                         "from Goal g " +
                         "where g.user = :user ", Goal.class)
                 .setParameter("user", user)
                 .getResultList();
+        return CheckStatus(goals);
+    }
+
+    private List<Goal> CheckStatus(List<Goal> goals) {
+        List<Goal> goalsToReturn = new ArrayList<>();
+        for (Goal goal : goals) {
+            if ((goal.getDeadlineDate().isBefore(LocalDate.now())
```

Comment:
Please extract `LocalDate.now()` to a variable and place it outside the `for` loop, to ensure consistent comparison of all goals against a single date and to reduce the amount of calls to `LocalDate.now()`.

Reply:
extracted thank you.